src/routes/getSegmentInfo.ts

```diff
@@ -44,12 +44,14 @@ async function handleGetSegmentInfo(req: Request, res: Response) {
         return false;
     }
     const DBSegments = await getSegmentsByUUID(UUIDs);
-    if (DBSegments === null || DBSegments === undefined) {
+    // all uuids failed lookup
+    if (DBSegments.length === 0) {
```

Comment:
Can `DBSegments` be null? If so, maybe change to `DBSegments?.length`.

Reply:
DBSegments can be 
```js
[ undefined ]
```
if the UUIDs are valid but `getSegmentsByUUID` returns no results. If they're all invalid, then DBSegments is 
```js
[]
```

Reply:
I switched the order to not to get an out of bounds error when looking up `DBSegments[0]`